pkg/controllers/cluster/import.go

```diff
@@ -254,15 +230,26 @@ func (i *importHandler) importCluster(cluster *fleet.Cluster, status fleet.Clust
 		return status, err
 	}
 
-	if err := i.deleteOldAgent(cluster, kc); err != nil {
```

Comment:
You can't switch the order here. Otherwise fleet-agent deployment will be deployed then deleted.

Reply:
I would prefer not touching this unless we need to. Darren probably had a reason to clean up agent first then apply with new agent. Keep the same order so that we can avoid any regression.

Reply:
Got it. Only thing that has changed is the re-organization of the method itself now